zapc/src/hir_builder/ty.rs

```diff
@@ -91,9 +114,16 @@ impl<'a> HirBuilder<'a> {
 		})
 	}
 
-	fn generics_one_range(&mut self, generics: &[AstGeneric]) -> Option<AstRange> {
+	fn generics_one_range(&mut self, segment: &AstWord, generics: &[AstGeneric]) -> Option<AstRange> {
 		if generics.len() > 1 {
-			// todo: report extra generics
+			self.report(Report::IncorrectGenericCount {
+				type_span: segment.span(),
```

Comment:
this wouldn't be the span of the entire type
imagine the type `buffer<a, b>` - this would only be the span for `buffer` when you really want it to be the span for the whole type